fixed mama code to work on raw data

manumanuk opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/canlib/33)
<!-- Reviewable:end -->

zx100x100 reviewed (changes requested):
Looks like the get_radi_info function on line 630 hasn't been updated, that needs to happen

also just to confirm the switch here is from a floating point value representation to the raw ADC value? 

It would be much nicer if you can simply accept a single 16 bit integer instead of two 8 bit ones representing the high and low portions of the value. Inside the function you would then break the single value out into the two bytes needed for the CAN message.

Finally not sure if you did, but make sure you actually run the unit tests and make sure they pass

manumanuk commented:
* Changed get_radi_info function
* The floating point was the raw ADC value, yes
* Changed build_radi_info function to accept 16 bit value (do you want me to change get_radi_info to output a 16 bit too?
* Unit tests pass

wendi-yu reviewed (commented):
code-wise this looks fine to me, I'll let Zach rereview on the board specifics though

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/33#-:-MeYWh2D-I9La4s8AiRz:b-c7duoc)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @zx100x100)



<!-- Sent from Reviewable.io -->

zx100x100 commented on can_common.c:
```diff
@@ -344,8 +343,8 @@ bool build_radi_info_msg(uint32_t timestamp,
-    output->data[4] = int_value;
-    output->data[5] = deci_value;
+    output->data[4] = (uint8_t) (adc_value >> 8); // Upper byte
+    output->data[5] = (uint8_t) (adc_value); // Lower byte
```
Adding a mask vs relying on the implicit truncation of casting it to a uint8_t would be nice

zx100x100 reviewed (approved):
Aside from the one comment on adding a mask looks good to me. if you can address that it would be appreciated, but I'm not going to set the review as request changes

BluCodeGH reviewed (approved):
> Reviewed 4 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/33)* status: all files reviewed, 1 unresolved discussion (waiting on @manumanuk)



<!-- Sent from Reviewable.io -->
